### PR TITLE
Allow consumes declarations in EDLooper

### LINKED ISSUE #2: Enable EDLooper to declare ED and ES consumes
https://github.com/cms-sw/cmssw/issues/31745

- [x] Make EDLooper to inherit from EDConsumerBase
- [x] Add unit tests

### LINKED ISSUE #31745: EDLooper can not declare any consumption of ED or ES products
Because it does not inherit from EDConsumerBase.

I'm not sure if we strictly need it to declare consumes for ED products, but for ES it could be useful (especially if we want to deprecate/remove the non-token API).

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

Resolves https://github.com/cms-sw/cmssw/issues/31745, resolves https://github.com/makortel/framework/issues/2.

#### PR validation:

Framework unit tests run.

### CODE REVIEW

**Comment by makortel** [comment]
code-checks

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@Dr15Jones Please review

File: FWCore/Framework/test/stubs/IntTestLooper.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          esToken_(esConsumes()),
+          valuesBeginRun_(iPSet.getUntrackedParameter<std::vector<int>>("expectBeginRunValues")),
+          valuesBeginLumi_(iPSet.getUntrackedParameter<std::vector<int>>("expectBeginLumiValues")),
+          valuesEvent_(iPSet.getUntrackedParameter<std::vector<int>>("expectEventValues")),
+          valuesEndLumi_(iPSet.getUntrackedParameter<std::vector<int>>("expectEndLumiValues")),
+          valuesEndRun_(iPSet.getUntrackedParameter<std::vector<int>>("expectEndRunValues")),
+          valueES_(iPSet.getUntrackedParameter<int>("expectESValue")) {}
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
```
Comment (Dr15Jones) [inline FWCore/Framework/test/stubs/IntTestLooper.cc:44]: Could be nice to say what the purpose of the looper is in a `setComment`.

File: FWCore/Framework/test/test_edlooper_consumes_cfg.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#Dummy looper will loop twice
+process.looper = cms.Looper("IntTestLooper",
+    srcEvent = cms.untracked.VInputTag("intEventProducer"),
+    expectEventValues = cms.untracked.vint32(42),
+    expectESValue = cms.untracked.int32(5)
+)
+
+process.p1 = cms.Path(process.intEventProducer)
+
+#process.Tracer = cms.Service("Tracer")
```
Comment (Dr15Jones) [inline FWCore/Framework/test/test_edlooper_consumes_cfg.py:32]: This can go away.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**